### ISSUE TITLE
Failing core-apiserver-proxy tests

### ISSUE THREAD

**Issue by dariadomagala** [kyma-project/kyma#3389]
**Description**
Apiserver-proxy tests are randomly failing with error:

```
'test-core-apiserver-proxy' has Failed status
Fetching logs from 'test-core-apiserver-proxy' with options tailLimit=2000 and bytesLimit=500000
KUBECTL_OUT should be 21230702-6129-401d-9810-e224fd9bea64
KUBECTL_OUT is
TEST FAILED
```

Prow: https://status.build.kyma-project.io/view/gcs/kyma-prow-logs/logs/post-master-kyma-gke-integration/1110889815403925504

**Expected result**

**Actual result**

**Steps to reproduce**

Run `testing.sh` script on a running cluster.

**Troubleshooting**

**Comment by Disper** [comment]
We should add more logs as the reason tests fails is not clear right now.

**Comment by sjanota** [comment]
I'm able to reproduce the problem. It happens once in a twenty runs maybe on my cluster. Now the debugging part.

**Comment by sjanota** [comment]
What I've found so far: 
1. The problem is that sometimes `kubectl exec` returns no output at all. I wasn't able to find out why.
2. I run the test wihtout apiserver-proxy (test was talking to apiserver using service account) and it worked perfectly fine. This means that the problem is within apiserver-proxy.
3. I've added some retries to the test and it also helped. Test haven't failed once.
4. I've added some delay between test file creation and `kubectl exec`. It hand't any effect at all.

As a workaround we may introduce retries in test, but there is definitely something in apiserver-proxy. Probably SPDY proxy is closing connections too early.
